game-backend/src/helper/gameObjects.ts

```diff
@@ -63,6 +63,20 @@ export const removeGameObject = (state: State, idToDelete: ObjectId): State => {
   };
 };
 
+export const getNearestFood = (state: State, unit: Unit): Food => {
```

Comment:
One short note before I review the rest of it: Could you move this to a `state-selector`? @mailaenderli has integrated this concept for aggregating data for the `GameContext`. This way we don't pollute internal helpers with logic we only require for the API.